.github/workflows/integration.yml

```diff
@@ -119,7 +115,7 @@ jobs:
           command: bash scripts/ci-integration-test.sh amqp
 
       - if: (github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.appsignal == 'true') &&
-          (github.event_name != 'pull_request' || needs.detect-user.outputs.IS_TEAM_MEMBER == 'true')
+          (github.event_name != 'pull_request' || env.PR_HAS_ACCESS_TO_SECRETS == 'true')
```

Comment:
We don't need to change this now, but I'm thinking we could have actually had each integration test step that needed access to a given secret to just check access for that particular secret.